datajoint/connection.py

```diff
@@ -102,7 +102,11 @@ def is_connected(self):
         """
         Returns true if the object is connected to the database server.
         """
-        return self._conn.ping()
+        try:
+            self._conn.ping()
```

Comment:
What exactly happens when no connection?

Reply:
If a stale connection is present, previously `ping` caused the re-establishment of the stale connection. This solution is not ideal as it would not work properly for previous versions of the PyMySQL's `ping`. To be compatible to both the new and old, use `try... return self.__conn.ping()` or require the new Pymysql version in the requirements.